src/nnet3/nnet-simple-component.cc

```diff
@@ -1171,6 +1171,118 @@ void RectifiedLinearComponent::StoreStats(
   StoreStatsInternal(out_value, &temp_deriv);
 }
 
+void EluComponent::Propagate(
+    const ComponentPrecomputedIndexes *indexes,
+    const CuMatrixBase<BaseFloat> &in,
+    CuMatrixBase<BaseFloat> *out) const {
+  // Apply exponential linear function (x >= 0 ? x : exp(x) - 1)
+  out->CopyFromMat(in);
+  CuMatrix<BaseFloat> exp_negated_relu(in);
```

Comment:
kPropagateInPlace is on so I will need to make a separate copy, right? Or do you want me to turn it off?